public/js/components/card-drop-down.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { Component, PropTypes } from 'react';
+import cx from 'classnames';
+
+import CardIcon from 'components/card-icon';
+import { gettext } from 'utils';
+
+const defaultSelectText = gettext('Please select');
+
+
+export default class CardDropDown extends Component {
+
+  static propTypes = {
+    cards: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.number,
+        resource_uri: PropTypes.string,
+        truncated_id: PropTypes.string,
+        type_name: PropTypes.string,
+      })
+    ).isRequired,
+    cssModifier: PropTypes.string,
+    onCardChange: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedText: defaultSelectText,
+      selectedCardType: null,
+      isFocused: false,
+    };
+  }
+
+  handleFocus = () => {
+    this.setState({isFocused: true});
+  }
+
+  handleBlur = () => {
+    this.setState({isFocused: false});
+  }
+
+  handleChange = e => {
+    if (this.props.onCardChange) {
+      this.props.onCardChange(e);
+    }
+    var selectedText;
+    var selectedNode = e.target.options[e.target.selectedIndex];
+    console.log(selectedNode);
+    if (selectedNode) {
+
+      var selectedCardType = selectedNode.getAttribute('data-type');
+      console.log('selectedCardType: ' + selectedCardType);
+      if (selectedCardType) {
+        selectedText = selectedNode.firstChild.nodeValue.replace(
```

Comment:
I'm pretty sure this is a strict no-no in react. I would assume instead you could just render the `{optionText}` differently once you receive `selectedCardType` in the state change, no? Did that not work?